### PR TITLE
Move java specific evaluators from OMR

### PR DESCRIPTION
**Pull request by dchopra001**
Many extern evaluators declarations in z/J9CodeGenerator.cpp are defined in OMR. These evaluators should be in OpenJ9 instead. As part of this commit, all extern declarations of evaluators are removed from J9CodeGenerator.cpp and function definitions are imported over from OMR where applicable.

Closes: #12911

### CODE REVIEW

**Comment by dchopra001** [comment]
@fjeremic Can I get a review please? 

See https://github.com/eclipse/omr/pull/6063 for related OMR PR. I think this can break the build. But hopefully we can time the merge towards EOD today so it doesn't block anyone.

**Review by fjeremic — APPROVED** [review]

**Comment by fjeremic** [comment]
Jenkins test sanity all jdk8,jdk11 depends eclipse/omr#6063

**Comment by fjeremic** [comment]
Jenkins test sanity all jdk8,jdk11 depends eclipse/omr#6063

**Comment by dchopra001** [comment]
The failing tests are related to actually compiling the Tests or prior to executing any tests.

**Comment by dchopra001** [comment]
@fjeremic can we try running the tests on the failing platforms again?

**Comment by fjeremic** [comment]
> @fjeremic can we try running the tests on the failing platforms again?

We'll merge this today. It's a Z only change and the failures are infra related on x86. No point waiting for those tests to re-run.

**Comment by fjeremic** [comment]
Just waiting for a good OMR-Acceptance build to pass before merging this to avoid build breaks. We have 8 commit difference now which is a bit too much. The builds have been a bit turbulent as well over the weekend and today with some build breaks so I'd rather wait the extra day.

**Comment by dchopra001** [comment]
> We have 8 commit difference now which is a bit too much.

If you want I can do another rebase to bring the branches (here and the related OMR PR) up to speed. We can still wait till whenever you feel is a good time to merge, but at least this way the branches stay fresh.

**Comment by fjeremic** [comment]
> If you want I can do another rebase to bring the branches (here and the related OMR PR) up to speed. We can still wait till whenever you feel is a good time to merge, but at least this way the branches stay fresh.

No need. I can force push the branches to avoid breaking builds or others picking up broken code. The state of the repos need to be good for that though. Let's just sit tight for an extra day.

**Pull request merged**
